### PR TITLE
ENT-3566: Refactor search common components out from learner portal so admin portal can also use it

### PR DESCRIPTION
**Pull request by binodpant**
Bringing in all course search related code under src/course-search here

in draft until following is fixed:

 - [X] npm run lint fails due to an error in no-cycles rule configuration https://edx-internal.slack.com/archives/CD4E4BCM9/p1605214473237100
 - [X] es5 check fails in vendor bundle
 - [X] need to use this dep locally in a cleaned up learner portal and ensure it works

### CODE REVIEW

File: src/course-search/data/utils.js
Comment (binodpant) [inline src/course-search/data/utils.js:1]: moving this file and test here since no other code in LP uses it right now

File: package.json
Comment (binodpant) [inline package.json:51]: these deps can go due to frontend-build

File: public/index.html
Comment (binodpant) [inline public/index.html:1]: this was needed by the fedx-scripts webpack command

File: README.rst
Comment (binodpant) [inline README.rst:12]: I will also likely update this version in frontend-build, ran this idea past @davidjoy  already but having the dep in our package.json unblocks this PR

File: .eslintrc.js
Comment (raq929) [inline .eslintrc.js:6]: I'm assuming this stuff makes it the same as `frontend-app-learner-portal-enterprise`? 
I wonder if we should use this file as a joint `.eslintrc` for all our enterprise repos so they are all on the same version of everything....

File: package.json
Comment (raq929) [inline package.json:13]: Yessss seems great

File: package.json
Comment (raq929) [inline package.json:92]: hahahaha we are putting them in only to remove them later with the rebrand. 😢 
(No issues, but I'll make a rebrand ticket for this repo now.)

File: src/hooks.js
Comment (raq929) [inline src/hooks.js:40]: I would actually change this to not be a default export anymore (you'll have to disable an eslint rule for the file; that's expected in hooks files). You'll have to change the imports to import `{ useEnterpriseConfig }`.

File: src/utils/tests.jsx
Comment (raq929) [inline src/utils/tests.jsx:5]: Hmmmm.... this is going to be interesting when you port it to the admin portal. Presumably it has a different AppContext. We might want to rename and / or check what the admin portal does, because I think `AppContext` should probably live in it's app, not in the shared repo. If they're similar enough I could be wrong, though.

**Review by raq929 — APPROVED** [review]
One small comment about your hooks file, and I thing you should take a look at the AdminPortal AppContext (if it exists, or find where it's getting the enterprise data that's used for search) before we merge, but otherwise, it's looking good!

File: src/utils/tests.jsx
Comment (binodpant) [reply src/utils/tests.jsx:5]: good point!

File: src/hooks.js
Comment (binodpant) [reply src/hooks.js:40]: good point!

File: src/utils/tests.jsx
Comment (binodpant) [reply src/utils/tests.jsx:5]: hmm @raq929 I don't see it being used at all! and now have this in both repos after refactor. Perhaps we can remove it..? I also did not find any usage of AppContext in admin portal repo

File: .eslintrc.js
Comment (binodpant) [reply .eslintrc.js:6]: interesting thought!

File: src/utils/tests.jsx
Comment (binodpant) [reply src/utils/tests.jsx:5]: will leave it here for now, removing from learner portal, the same function for now

File: README.rst
Comment (adamstankiewicz) [reply README.rst:12]: @binodpant I can look into upgrading eslint-plugin-import in frontend-build, or review any PRs for it. I'm overdue to look at some of the upgrade PRs for frontend-build as the primary technical "owner" :)

File: package.json
Comment (adamstankiewicz) [inline package.json:18]: Should we make frontend-enterprise consistent with our other edX libraries (e.g., Paragon, frontend-platform), which don't have the is-es5 check as a script or called in .travis.yml?

File: package.json
Comment (adamstankiewicz) [reply package.json:92]: Thanks @raq929!

File: src/utils/tests.jsx
Comment (adamstankiewicz) [reply src/utils/tests.jsx:5]: Admin portal does not use frontend-build or frontend-platform right now, so it does not utilize AppContext.

**Review by adamstankiewicz — APPROVED** [review]
:shipit:

File: package.json
Comment (binodpant) [reply package.json:18]: yeah makes sense if the MFEs do it. George will remove travis.yml file as well

**Pull request merged**
